docs/dg/dev/architecture/coding-conventions.md

```diff
@@ -0,0 +1,2165 @@
+│           │   └── [Module]To[Module]ClientInterface.php
+│           └── Service
+│               ├── [Module]To[Module]ServiceBridge.php
+│               └── [Module]To[Module]ServiceInterface.php      
+├── Glue
+│   └── [Module]    
+│       └── Dependency
+│           ├── Client
+│           │   ├── [Module]To[Module]ClientBridge.php
+│           │   └── [Module]To[Module]ClientInterface.php
+│           └── Service
+│               ├── [Module]To[Module]ServiceBridge.php
+│               └── [Module]To[Module]ServiceInterface.php    
+└── Client
+    └── [Module]      
+        └── Dependency
+            ├── Client
+            │   ├── [Module]To[Module]ClientBridge.php
+            │   └── [Module]To[Module]ClientInterface.php
+            └── Service
+                ├── [Module]To[Module]ServiceBridge.php
+                └── [Module]To[Module]ServiceInterface.php  
+```
+
+**Description**
+
+According to the Interface Segregation Principle, every module defines an interface for each external class it relies upon. To decouple and encapsulate these dependencies effectively, we utilize `Bridges`. These `Bridges` serve to wrap dependencies such as [Facade](#facade), [Client](#client-facade), [Service](#service-facade), external library classes, and more. By doing so, a module explicitly declares its own requirements, enabling any class that implements the interface to be utilized.
+
+Challenges arise when a facade from another module seeks to implement the friend facade interface. This scenario introduces cross-module coupling between the facade and the implemented friend facade interface. To mitigate this issue, we employ the `Bridge pattern`. This approach facilitates seamless module decoupling, allowing both modules to evolve independently while maintaining a high degree of flexibility and maintainability.
+
+**Conventions**
+
+- The bridge class MUST define and implement an interface (`[CurrentModule]To[FacadeOwnerModule][Service|Client|Facade]Interface.php`) that holds the specification of each `public` method (mind the missing bridge word!).
+- `Bridges` MUST contain only the delegation logic to the friend facade method.
+- The `Bridge` constructor MUST NOT contain parameters to avoid coupling to a specific class.
+- `Bridge` classes and interfaces MUST be declared as strict as possible for input arguments and returns values, compared to their friend method with valid types (no mixed, no x|y).
+
+**Guidelines**
+- Bridge vs adapter: for simplification, we keep using bridge pattern even when adapting the earlier version of a SCOS facade. Adapters are used when the remote class' life-cycle is independent to SCOS or there is a huge technical difference between the adaptee and adaptor.
+- During `Bridge` definitions, type definition mistakes in remote facades become more visible. In these cases, be aware of the cascading effect of changing/restricting an argument type in public API when you consider such changes.
+
```

Comment:
```suggestion
<details><summary markdown='span'>Additional Conventions for Project Development</summary>
- We strongly discourage the usage of bridges in projects unless necessary due to business logic.
</details>
```

Reply:
You are in Module contribution section, nothing is applicable to project development.